components/scream/src/physics/p3/atmosphere_microphysics.cpp

```diff
@@ -138,36 +205,105 @@ void P3Microphysics::run_impl (const Real dt)
     Kokkos::deep_copy(m_p3_host_views_out.at(it.first),it.second.get_view());
   }
 
-  // Call f90 routine
-  p3_main_f90 (dt, m_raw_ptrs_in["zi"], m_raw_ptrs_in["pmid"], m_raw_ptrs_in["dp"], m_raw_ptrs_in["ast"], m_raw_ptrs_in["ni_activated"], m_raw_ptrs_in["nc_nuceat_tend"], m_raw_ptrs_out["q"], m_raw_ptrs_out["FQ"], m_raw_ptrs_out["T"], m_raw_ptrs_out["qv_prev"], m_raw_ptrs_out["T_prev"]);
-
   // Copy outputs back to device
   for (auto& it : m_p3_fields_out) {
     Kokkos::deep_copy(it.second.get_view(),m_p3_host_views_out.at(it.first));
   }
```

Comment:
Should your call to `auto elapsed_microsec = P3F::p3_main()` below come before you copy outputs back to device?

Reply:
I believe that p3_main acts on device.  So we will need to make sure that host and device views are synced before calling p3_main.  Here I am removing the F90 version, which happened on host and replacing with the C++ version.  But, I may be misunderstanding how things work, it wouldn't be the first time, so I am open to some pushback.

Reply:
Well, elapsed_microsec is not even used, so maybe we can remove it altogether?

Reply:
ahhh I see, your comment is about `elapsed_microsec`.  It's true it isn't used, so @bartgol 's point is well-taken.  I'm not opposed to removing it, but that would require touching the unit testing code as well, which is maybe outside the scope of this PR (?)

Reply:
I am not sure why it would touch the UT code. TBC, I'm not suggesting to make p3_main return void. I'm just suggesting we don't need the elapsed_microsec var here.

Reply:
Is it possible to call a routine that returns `non-void` without assigning it some named output variable?

Reply:
Yep. Just omit the assignment.

Reply:
oh cool, well, that's been done.